### PR TITLE
Add threat intel source apis for create delete get search operations

### LINKED ISSUE #7714: [DOC] Security Analytics - Need documentation for Threat Intel Analytics platform in the Security Analytics plugin
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.
We are adding threat intel analytics platform as part of the security anayltics plugin. We require documentaiton of the features and APIs
***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
 2.16 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

https://github.com/opensearch-project/security-analytics/issues/1117

### PR DESCRIPTION
**Pull request by eirsep**
### Description
added threat intel source apis for create delete get search operations

### Issues Resolved
Closes [#7714](https://github.com/opensearch-project/documentation-website/issues/7714)  

### Version
2.16

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by eirsep** [comment]
@AWSHurneyt plz review

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+   "period": "10",
+   "unit": "DAYS"
+  }
+ },
+ "source": {
+  "s3": {
+   "bucket_name": "threat-intel-s3-test-bucket",
+   "object_key": "alltypess3object",
+   "region": "us-west-2",
+   "role_arn": "arn:aws:iam::248279774929:role/threat_intel_s3_test_role"
```
Comment (AWSHurneyt) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:201]: Should we instead use a placeholder here like `<ROLE_ARN>`?

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 2]
```diff
@@ -0,0 +1,637 @@
+---
+layout: default
+title: Threat Intelligence Source Configuration APIs
+parent: API tools
+nav_order: 50
+---
+
+# Threat Intelligence Source Configuration APIs
+
+The following APIs can be used for tasks related to threat intelligence source configurations.
```
Comment (AWSHurneyt) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:10]: @eirsep  Would it be helpful to define what a `source` is for our purposes?

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 3]
```diff
@@ -0,0 +1,637 @@
+---
+layout: default
+title: Threat Intelligence Source Configuration APIs
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:3]:
```suggestion
title: Threat intelligence source APIs
```
Can we shorten this to Threat intelligence source APIs or Thread intelligence APIs? See the other headings in this section.

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 4]
```diff
@@ -0,0 +1,637 @@
...
+parent: API tools
+nav_order: 50
+---
+
+# Threat Intelligence Source Configuration APIs
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:8]:
```suggestion
# Threat intelligence source APIs
```
(or same as above title)

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+
+The following APIs can be used for tasks related to threat intelligence source configurations.
+---
+## Create Threat intel source API
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:12]:
```suggestion
## Create threat intelligence source API
```
(Our headings are sentence case.)

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md (hunk 2 above)
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:10]:
```suggestion
You can use threat intelligence source API to configure tasks related to threat intelligence sources.
```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 6]
```diff
@@ -0,0 +1,637 @@
...
+layout: default
+title: Threat Intelligence Source Configuration APIs
+parent: API tools
+nav_order: 50
+---
+
+# Threat Intelligence Source Configuration APIs
+
+The following APIs can be used for tasks related to threat intelligence source configurations.
+---
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:11]:
```suggestion

```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 7]
```diff
@@ -0,0 +1,637 @@
...
+## Create Threat intel source API
+
+Creates a new threat intel source.
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:14]:
```suggestion
The following command creates a new threat intelligence source.
```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 8]
```diff
@@ -0,0 +1,637 @@
...
+
+```
+POST _plugins/_security_analytics/threat_intel/sources
+```
+
+### Request fields
+
+You can specify the following fields when creating a threat intel source.
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:22]:
```suggestion
The following fields are used to create a threat intelligence source.
```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 9]
```diff
@@ -0,0 +1,637 @@
...
+ | `description`                         | String           | A description of the threat intelligence source.                                                                                                                   |
+ | `enabled`                             | Boolean          | Indicates whether scheduled refresh of iocs from source is enabled i.e. the threat intel iocs downloaded from the source should be refreshed on a schedule or not. |
+ | `ioc_types`                           | Array of Strings | The STIX2 types of IOCs that the source supports. Supported ioc types: `["hashes", "domain-name","ipv4-addr", "ipv6-addr"]`.                                       |
+ | `source`                              | Object           | Contains the source details for the threat intelligence data.                                                                                                      |
+ | `source.ioc_upload`                   | Object           | Details for IOC upload. Applicable for `IOC_UPLOAD` type                                                                                                           |
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:35]:
```suggestion
 | `source.ioc_upload`                   | Object           | Details for IOC upload. Applicable for the `IOC_UPLOAD` type.                                                                                                           |
```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 10]
```diff
@@ -0,0 +1,637 @@
...
+ | `source.ioc_upload.file_name`         | String           | The name of the file containing IOCs. Example: `test`.Applicable for `IOC_UPLOAD` type                                                                             |
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:36]:
```suggestion
 | `source.ioc_upload.file_name`         | String           | The name of the file containing IOCs. Example: `test`. Applicable for the`IOC_UPLOAD` type.                                                                             |
```

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md [hunk 11]
```diff
@@ -0,0 +1,637 @@
...
+ | `description`  | String           | A description of the IOC. Example: `first one here`.                                                                                                 |
+ | `labels`       | Array of Strings | Labels or tags associated with the IOC. Example: `["ik"]`.                                                                                           |
+ | `feed_id`      | String           | A unique identifier for the feed to which the IOC belongs. Example: `jl`.                                                                            |
+ | `spec_version` | String           | The specification version used for the IOC. Example: `gavvnespe`.                                                                                    |
+ | `version`      | Integer          | A version number for the IOC. Example: `-4356924786557562654`.                                                                                       |
+
+### Response Fields
+
+| Field                                       | Type              | Description                                                                                                     |
```
Comment (hdhalter) [inline _security-analytics/api-tools/threat-intel/threat-intel-source.md:63]:
```suggestion
| Field                                       | Data type              | Description                                                                                                     |
```

**Review by hdhalter — COMMENTED** [review]
Nice job, @eirsep! I entered a few suggestion so you can see what we are looking for in a more thorough doc review. Thanks.

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md (hunk 2 above)
Comment (eirsep) [reply _security-analytics/api-tools/threat-intel/threat-intel-source.md:10]: Started off with the APIs
Once that's done, similar to detectors, I will define entities and detail them out in another commit to this PR

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md (hunk 1 above)
Comment (eirsep) [reply _security-analytics/api-tools/threat-intel/threat-intel-source.md:201]: Good point. It's a dummy aws account but  will let @Naarcha-AWS and @hdhalter recommned the right practice.

**Comment by eirsep** [comment]
@AWSHurneyt what documentation should we add related to the S3 Role ARN on what needs to be configured?

**Comment by eirsep** [comment]
@AWSHurneyt @Naarcha-AWS plz review

File: _security-analytics/api-tools/threat-intel/threat-intel-source.md (hunk 2 above)
Comment (eirsep) [reply _security-analytics/api-tools/threat-intel/threat-intel-source.md:10]: added a page for Threat intel overview plz check

**Comment by Naarcha-AWS** [comment]
Closing in favor of #7905
